middleware.go

```diff
@@ -124,71 +137,87 @@ func NewWithConfig(logger *slog.Logger, config Config) func(http.Handler) http.H
 
 			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
 			defer func() {
-				// headers := ww.Header()
-				// length := ww.BytesWritten()
-
-				// status := r.Response.StatusCode
 				status := ww.Status()
 				method := r.Method
 				route := chi.RouteContext(r.Context()).RoutePattern()
 				end := time.Now()
 				latency := end.Sub(start)
-				// ip := "x.x.x.x"
 				userAgent := r.UserAgent()
 
-				attributes := []slog.Attr{
+				rqAttributes := []slog.Attr{
 					slog.Time("time", end),
 					slog.Duration("latency", latency),
 					slog.String("method", method),
 					slog.String("path", path),
 					slog.String("route", route),
 					slog.Int("status", status),
-					// slog.String("ip", ip),
 					slog.String("user-agent", userAgent),
 				}
 
+				if config.WithRequestIP {
+					rqAttributes = append(rqAttributes, slog.String("ip", r.RemoteAddr))
+				}
+
 				if config.WithRequestID {
-					attributes = append(attributes, slog.String("request-id", middleware.GetReqID(r.Context())))
```

Comment:
Lets move `request-id` to a global `var RequestIDKey = "request-id"`

Reply:
I'm a bit torn on the kebab case naming of `request-id`, `trace-id` & `span-id`, but tbh I don't really have anything better except just shortening `request-id` to `id` since it's already nested in the `request` group